### PR TITLE
#37 Prevents the immediate logout->login loop

### LINKED ISSUE #37: Automatic re-login if you try to logout
If you configure your security Realm to use "Logged-in users can do anything" and disable anonymous read access, you effectively can't logout.
The following is what happens:

1. Logout
2. Jenkins detects that you're not logged in, so redirects you to /securityRealm/commenceLogin
3. commenceLogin redirects you to your IdP
4. You are already logged in with your IdP and have a cookie, so it automatically (if there is no approval screen or your preferences are cached - which is the standard case) redirects you back to /securityRealm/finishLogin
5. You're logged in again.

### PR DESCRIPTION
**Pull request by agentgonzo**
Fixes #37

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java [hunk 1]
```diff
@@ -598,9 +599,11 @@ private String buildOAuthRedirectUrl() throws NullPointerException {
     }
 
     /**
-    * This is where the user comes back to at the end of the OpenID redirect ping-pong.
+     * This is where the user comes back to at the end of the OpenID redirect ping-pong.
+     * @param request The user's request
+     * @return an HttpResponse
     */
-    public HttpResponse doFinishLogin(StaplerRequest request) throws IOException {
```
Comment (mjmbischoff) [inline src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java]: see https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Foic-auth-plugin/detail/PR-38/2/pipeline

Comment (agentgonzo) [reply src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java]: Fixed now.

**Comment by agentgonzo** [comment]
@mjmbischoff - is this of interest to you? It's been sat dangling for a while ;-)

**Comment by mjmbischoff** [comment]
Ah damn, thought I already merged this. 

I was planning to do a release in which case I would have noticed it sooner, but things keep coming up. Thank you for the gentle nudge ;-)

**Pull request merged**